components/datetime/src/provider/neo.rs

```diff
@@ -544,6 +545,131 @@ pub struct DateTimePatternV1<'data> {
     pub pattern: runtime::GenericPattern<'data>,
 }
 
+// conceptually:
+// {
+//   has_long: bool,
+//   has_medium: bool,
+//   has_short: bool,
+//   index: u16, // index of first pattern (long if present, else medium, else short)
+// }
+#[allow(missing_docs)] // TODO
+#[derive(Debug, Copy, Clone, PartialEq)]
+#[cfg_attr(
+    feature = "datagen",
+    derive(serde::Serialize, databake::Bake),
+    databake(path = icu_datetime::provider::neo),
+)]
+#[cfg_attr(feature = "serde", derive(serde::Deserialize))]
+pub struct SkeletonDataIndex {
+    /// If true, the first pattern is for `Long`.
+    /// If false, fall back to the next pattern (`Medium``).
+    pub has_long: bool,
+    /// If true, the next pattern is for `Long`.
+    /// If false, fall back to the next pattern (`Short`).
+    pub has_medium: bool,
```

Comment:
question: I was under the impression that in the original model we could also set `has_short` to false?

I guess in that case there's still something that `short` falls back to, so we can instead call that `short`

Reply:
I don't want all of them to be `false` because that's an error case we need to handle. I don't remember why we allowed that case. It seems cleaner to have long inherit medium and medium inherit short, which can never fail.

Reply:
wfm